### PR TITLE
Weather Card

### PR DESCRIPTION
**Pull request by Aaaaa501**
Created a weather card using openmetro weather API (https://open-meteo.com/).
Card looks like this:
Screen Shot 2023-06-25 at 11 08 46 PM

### CODE REVIEW

File: src/components/cards/weatherCard.vue [hunk 1]
```diff
@@ -0,0 +1,177 @@
+<template>
+  <card v-if="weatherData != null" class="card">
```
Comment (JosephShepin) [inline src/components/cards/weatherCard.vue:2]: can remove `!= null`

File: src/components/cards/weatherCard.vue [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  data() {
+    return {
+      icons: { faCloud, faTint, faSun },
+      weatherData: null,
+    };
+  },
+  mounted() {
+    let currentDate = new Date();
+    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+    function formatDate(date) {
```
Comment (JosephShepin) [inline src/components/cards/weatherCard.vue:66]: add a return type here and can remove eslint disable line

File: src/components/cards/weatherCard.vue [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+          const daily = {
+            date: idate,
+            temp_high: data.daily.temperature_2m_max[i],
+            temp_low: data.daily.temperature_2m_min[i],
+            rain_percent: data.daily.precipitation_probability_max[i],
+            cloudcover: avgCloudCover,
+          };
+          dailyData.push(daily);
+        }
+        console.log(dailyData)
```
Comment (JosephShepin) [inline src/components/cards/weatherCard.vue:97]: remove console logs

File: src/components/cards/weatherCard.vue [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        this.weatherData = dailyData;
+      })
+      .catch((error) => {
+        console.error('Error fetching weather data:', error);
```
Comment (JosephShepin) [inline src/components/cards/weatherCard.vue:101]: also remove console logs. If it fails to fetch, let's hide the card

**Review by JosephShepin — CHANGES_REQUESTED** [review]
build failing

**Comment by Aaaaa501** [comment]
It should work now. Any other issues?

**Comment by JosephShepin** [comment]
Screenshot 2023-06-25 at 11 39 08 PM
do we need this badge?

**Comment by JosephShepin** [comment]
is there a way we can show less days? I feel like it's not needed to show 6 days in advance

**Comment by Aaaaa501** [comment]
Yeah it would be pretty simple, but how many days do you think is best?

**Comment by JosephShepin** [comment]
i also feel like we can refine the ui here. i found some ideas on dribble that could help out. I think the icons can be switched out with something more flat looking like the ionicons. I think we can get rid of the "F" for Fahrenheit - everyone uses Fahrenheit so I think it's assumed and just leaving the degree symbol. We could also add a setting for it if we really think people will use celcius. Would it be possible to refine the card to a design similar to the ones below? I think it would be easier to read look cooler. 

Screenshot 2023-06-25 at 11 42 57 PM
Screenshot 2023-06-25 at 11 44 28 PM

weather ionicons might look better:
Screenshot 2023-06-25 at 11 45 12 PM

**Comment by JosephShepin** [comment]
Also we are going to want to cache these responses in local storage so we reduce our api calls. Maybe a 6 hour timeout or something around that should be good.

**Comment by aw-0** [comment]
Great work, thanks for the PR! On top of the existing comments from joey, would also like to recommend having all of the temperature `<p>`'s be centered for consistency.
CleanShot 2023-06-26 at 17 44 35

Additionally--

> Also we are going to want to cache these responses in local storage so we reduce our api calls. Maybe a 6 hour timeout or something around that should be good.

Could even be possible to do something similar to the calendar where we have it fetch weather data daily (aka on build time) with the calendar updates? Just an idea

**Comment by Aaaaa501** [comment]
On the weather ionicons, im not sure how to add ionicons into a vue project. Do you know how?

**Comment by aw-0** [comment]
Love those changes! The only other comment I may have is making the current day "Today" instead of the name of the day, causes less confusion when glancing at the UI.

Not sure if joey can comment but I haven't personally worked with ionicons, but looks like there's a [NPM package for Vue](https://www.npmjs.com/package/vue-ionicons). If that doesn't work, here's a [DIY article](https://dev.to/devmount/how-to-use-ionicons-v5-with-vue-js-53g2) you could try that manually adds support through custom elements without a package.

**Comment by Aaaaa501** [comment]
Ive added the new features, is the current version good?

Also
> Screenshot 2023-06-25 at 11 39 08 PM do we need this badge?
Yes, I read on the website that they require attribution for usage

**Comment by JosephShepin** [comment]
This looks really nice. One thing I noticed is that the temperatures are in Celsius - can we make it Fahrenheit?

File: src/components/cards/WeatherCard.vue [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    title: { type: String, default: 'Weather' },
+  },
+  data() {
+    return {
+      icons: { faCloud, faTint, faSun },
+      weatherData: null,
+    };
+  },
+  mounted() {
+    // localStorage.getItem('weatherDataCache')
```
Comment (JosephShepin) [inline src/components/cards/WeatherCard.vue:55]: i think we can get rid of this line?

**Review by JosephShepin — CHANGES_REQUESTED** [review]
this is really close to being done

**Review by JosephShepin — APPROVED** [review]
Thanks so much! Really nice work! 🥳

**Comment by Aaaaa501** [comment]
Does the pull request automatically merge?

**Comment by JosephShepin** [comment]
@Aaaaa501 feel free to merge whenever

**Comment by Aaaaa501** [comment]
I don’t have write access

**Pull request merged**